atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -318,62 +152,32 @@ def sort_annotations(self):
-                    continue
-
-                platforms = sorted_digests.setdefault(hostname, {})
-                repos = platforms.setdefault(plat, [])
-                repos.append(digest)
-
-        sources = {}
-        for registry in self.registries:
-            hostname = registry_hostname(registry)
-            platforms = sorted_digests.get(hostname, {})
-
-            if set(platforms) != all_platforms:
-                raise RuntimeError("Missing platforms for registry {}: found {}, expected {}"
-                                   .format(registry, sorted(platforms), sorted(all_platforms)))
-
-            selected_digests = {}
-            for p, repos in platforms.items():
-                selected_digests[p] = sorted(repos, key=lambda d: d['repository'])[0]
-
-            sources[registry] = selected_digests
-
-        return sources
-
-    def get_registry_session(self, registry):
-        registry_conf = self.registries[registry]
-
-        insecure = registry_conf.get('insecure', False)
-        secret_path = registry_conf.get('secret')
-
-        return RegistrySession(registry, insecure=insecure,
-                               dockercfg_path=secret_path,
-                               access=('pull', 'push'))
+        return self.manifest_util.sort_annotations(all_annotations)
 
     def run(self):
-        digests = dict()
+        list_json = ""
+        # No other plugin really needs the list of repos, but we should return it for
+        # unit tests
+        repos = []
+
         for registry, source in self.sort_annotations().items():
-            session = self.get_registry_session(registry)
+            session = self.manifest_util.get_registry_session(registry)
 
             if self.group:
-                repo, digest = self.group_manifests_and_tag(session, source)
+                repo, digest, list_json = self.group_manifests_and_tag(session, source)
                 self.log.debug("repo: %s digest: %s", repo, digest)
-                digests[repo] = digest
+                repos.append(repo)
             else:
                 found = False
                 if len(source) != 1:
                     raise RuntimeError('Without grouping only one source is expected')
                 for digest in source.values():
-                    self.tag_manifest_into_registry(session, digest)
+                    source_digest = digest['digest']
+                    source_repo = digest['repository']
+                    self.manifest_util.tag_manifest_into_registry(session, source_digest,
+                                                                  source_repo,
+                                                                  self.workflow.tag_conf.images)
                     found = True
                 if not found:
                     raise ValueError('Failed to find any platform')
-        return digests
+        return {'digest': digest, 'repos': repos, 'manifest_list': list_json}
```

Comment:
why do we need repos here?

we should return digest, digest_type (or both of them in some dict), and list_json

if we won't return digest_type ( we would need to get_manifest if list_json wasn't specified, because if list_json is specified we know it is list.v2)

Reply:
there's supposed to be a comment that it was testing purposes, but it turns out to be not be very useful.

Reply:
we are looping through floating tag images in push_floating_images anyway so we know target_repo

Reply:
ic :)

Reply:
So we'll return a ManifestDigest (which includes the type as well as the digest), and the manifest list text.